tests/api_workflow/test_api_workflow_tags.py

```diff
@@ -1,89 +1,214 @@
-import os
-import tempfile
-import unittest
-import warnings
-
-import numpy as np
-
-import lightly
-from lightly.openapi_generated.swagger_client import SamplingMethod
-from lightly.openapi_generated.swagger_client.models.tag_data import TagData
-from tests.api_workflow.mocked_api_workflow_client import (
-    MockedApiWorkflowClient,
-    MockedApiWorkflowSetup,
-)
-
-
-class TestApiWorkflowTags(MockedApiWorkflowSetup):
-    def setUp(self) -> None:
-        lightly.api.api_workflow_client.__version__ = lightly.__version__
-        warnings.filterwarnings("ignore", category=UserWarning)
-        self.api_workflow_client = MockedApiWorkflowClient(token="token_xyz")
-
-        self.valid_tag_name = self.api_workflow_client.get_all_tags()[0].name
-        self.invalid_tag_name = "invalid_tag_name_xyz"
-        self.valid_tag_id = self.api_workflow_client.get_all_tags()[0].id
-        self.invalid_tag_id = "invalid-tag_id_xyz"
-
-    def tearDown(self) -> None:
-        warnings.resetwarnings()
-
-    def test_get_all_tags(self):
-        self.api_workflow_client.get_all_tags()
+from typing import List, Optional
+
+import pytest
+from pytest_mock import MockerFixture
+
+from lightly.api import ApiWorkflowClient
+from lightly.api.api_workflow_tags import TagDoesNotExistError
+from lightly.openapi_generated.swagger_client.models import TagCreator, TagData
+from tests.api_workflow.utils import generate_id
+
+
+def _get_tags(
+    dataset_id: str, tag_name: str = "just-a-tag", prev_tag_id: Optional[str] = None
+) -> List[TagData]:
+    return [
+        TagData(
+            id=generate_id(),
+            dataset_id=dataset_id,
+            prev_tag_id=prev_tag_id,
+            bit_mask_data="0x5",
+            name=tag_name,
+            tot_size=4,
+            created_at=1577836800,
+            changes=[],
+        )
+    ]
+
+
+def test_create_tag_from_filenames(mocker: MockerFixture) -> None:
+    dataset_id = generate_id()
+    tags = _get_tags(dataset_id=dataset_id, tag_name="initial-tag")
+    mocker.patch.object(ApiWorkflowClient, "__init__", return_value=None)
+    mocker.patch.object(ApiWorkflowClient, "get_all_tags", return_value=tags)
+    mocked_get_filenames = mocker.patch.object(
+        ApiWorkflowClient, "get_filenames", return_value=[f"file{i}" for i in range(3)]
+    )
+    mocked_api = mocker.MagicMock()
+
+    client = ApiWorkflowClient()
+    client._tags_api = mocked_api
+    client._dataset_id = dataset_id
+    client._creator = TagCreator.UNKNOWN
+    client.create_tag_from_filenames(fnames_new_tag=["file2"], new_tag_name="some-tag")
+    mocked_get_filenames.assert_called_once()
+    mocked_api.create_tag_by_dataset_id.assert_called_once()
+    kwargs = mocked_api.create_tag_by_dataset_id.call_args[1]
```

Comment:
FYI, you could use `kwargs = mocked_api.create_tag_by_dataset_id.call_args.kwargs` instead.